libcaf_core/caf/config_option.hpp

```diff
@@ -105,6 +105,9 @@ class CAF_CORE_EXPORT config_option {
   /// - Store the converted value unless this option is stateless.
   error sync(config_value& x) const;
 
+  /// Tries to extract a value from the set location.
+  config_value get() const;
```

Comment:
This is a very dangerous getter that we should remove again. The internal pointer may be null and if it isn't, it's pointing to something that the user has given us, so there's no need to ask the config option. What was the motivation to add this?

Reply:
Currently, there is no public interface to use the `get_impl()` function and this getter has been added to use the implementation. I am removing the getter and the related tests.

Reply:
What `get_impl`? Is it something `meta_->get(value_)` is calling internally? Then we can simply test it via the meta objects directly.

Reply:
Yes `meta_->get(value_)` calls the `get_impl()` which is defined in `make_config_option.hpp` https://github.com/actor-framework/actor-framework/blob/e876df5f0797206bdb80b79f442db38285f16d56/libcaf_core/caf/make_config_option.hpp#L33-L38

I am trying to write tests with meta objects directly.

Reply:
We can not use the meta objects directly as we need the object that is referenced by the following internal function https://github.com/actor-framework/actor-framework/blob/e876df5f0797206bdb80b79f442db38285f16d56/libcaf_core/caf/make_config_option.hpp#L40-L45

I think that the tests are not necessary here as the function pointer will be removed eventually
https://github.com/actor-framework/actor-framework/blob/e876df5f0797206bdb80b79f442db38285f16d56/libcaf_core/caf/config_option.hpp#L48-L50

Reply:
Thanks for digging into this. I've removed the obsolete pieces. 👍